src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -26,5 +27,187 @@ public partial class AppDomain
                 AppContext.UnhandledException -= value;  
             }  
         }  
+
+        public string DynamicDirectory => null;
+        public string FriendlyName
+        {
+            get
+            {
+                Assembly assembly = Assembly.GetEntryAssembly();
+                string name = null;
+                if(assembly == null)
+                    name = "DefaultDomain";
+                else
+                    name = assembly.GetName().Name;
+                return name;             
+            }
+        }
+        public int Id => 1;
+        public bool IsFullyTrusted => true;
+        public bool IsHomogenous => true;
+        public event System.EventHandler DomainUnload { add { } remove { } }
+
+        public event System.EventHandler<System.Runtime.ExceptionServices.FirstChanceExceptionEventArgs> FirstChanceException
+        {  
+            add  
+            {  
+                AppContext.FirstChanceException += value;  
+            }  
+  
+            remove  
+            {  
+                AppContext.FirstChanceException -= value;  
+            }  
+        }  
+
+        public event System.EventHandler ProcessExit
+        {  
+            add  
+            {  
+                AppContext.ProcessExit += value;  
+            }  
+  
+            remove  
+            {  
+                AppContext.ProcessExit -= value;  
+            }  
+        }  
+
+        public string ApplyPolicy(string assemblyName)
+        {
+            if (assemblyName == null)
+                throw new ArgumentNullException("assemblyName");
+            if (assemblyName.Length == 0) // String.Empty
+                throw new ArgumentException("assemblyName");
+            return assemblyName;
+        }
+
+        public static System.AppDomain CreateDomain(string friendlyName) 
+        { 
+            if(friendlyName == null) throw new ArgumentNullException("friendlyName");
+            throw new PlatformNotSupportedException(); 
+        }
+
+        public int ExecuteAssembly(string assemblyFile) => ExecuteAssembly(assemblyFile, null, null, System.Configuration.Assemblies.AssemblyHashAlgorithm.None);
+        public int ExecuteAssembly(string assemblyFile, string[] args) => ExecuteAssembly(assemblyFile, args, null, System.Configuration.Assemblies.AssemblyHashAlgorithm.None);
+        public int ExecuteAssembly(string assemblyFile, string[] args, byte[] hashValue, System.Configuration.Assemblies.AssemblyHashAlgorithm hashAlgorithm)
+        {
+            if(assemblyFile == null)
+                throw new ArgumentNullException("assemblyFile");
+            /* TODO Assembly.LoadFrom is not implemented yet
```

Comment:
It would be useful to have link to the issue that tracks the work to get it finished next to all TODOs.